Restore public documentation with key information for Firecracker users (#4423)
firecracker-microvm/firecracker: docs/tracing.md

## Introduction

> Firecracker implements a framework for instrumentation based tracing with the

Reviewer: commit `chore(docs) Update tracing.md`:
- fix -> fixes
- refine -> refines

Author: done, thanks

> aim to improve its debug-ability.

Reviewer: not sure the hyphen is needed.

Firecracker implements instrumentation based tracing via
[`log`](https://github.com/rust-lang/log) and
[`log_instrument`](../src/log-instrument), outputting a `Trace` level log when
entering and exiting every function.

> Enabling tracing impact Firecracker binary size and its performance, so it is

Reviewer: I don't think this is correct. Enabling/disabling tracing is a runtime action and can't change FC binary size. Did you mean adding instrumentation? It does indeed change the binary size.
However I'm not quite sure about the original "It is disabled by default at compile-time" either. Maybe it can be reworded as "tracing is disabled by default"?

Reviewer 2: I've posted a suggestion comment https://github.com/firecracker-microvm/firecracker/pull/4423/commits/ded99cf417323fb18222b704acea6b7cc6a1dc25#r1478237955 (GitHub doesn't allow them in comment chains it seems)

> Adding traces impacts Firecracker binary size and its performance, so tracing is
> disabled by default. Tracing functionality is also disabled on the available
> release binaries.

Reviewer 2: Related to https://github.com/firecracker-microvm/firecracker/pull/4423#discussion_r1477824526 I would suggest:

```suggestion
Tracing is disabled by default as  instrumentation impacts Firecracker's binary size.
Tracing functionality is disabled on the release binaries.
```

Author: I modified (almost) as you suggested, thanks.
